clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -155,7 +155,10 @@ public void run() {
             this.accumulator.abortIncompleteBatches();
         }
         try {
-            this.client.close();
+            if (!forceClose)
+                this.client.closeGracefully();
```

Comment:
Do we actually need this? It seems that we wait until `inFlightRequestCount == 0` and there are no unsent data in the accumulator.